src/modules/core/components/SingleUserPicker/utils.ts

```diff
@@ -7,20 +7,17 @@ export const filterUserSelection = (data, filterValue) => {
     return data;
   }
 
-  const filtered = data.filter(
-    (user) =>
-      user &&
-      user.profile &&
-      user.profile.username &&
-      filterValue &&
-      (user.profile.username
-        .toLowerCase()
-        .includes(filterValue.toLowerCase()) ||
-        user.profile.walletAddress
-          .toLowerCase()
-          .includes(filterValue.toLowerCase()) ||
-        user.id.toLowerCase().includes(filterValue.toLowerCase())),
-  );
+  const filtered = data.filter((user) => {
+    const { username, walletAddress } = user?.profile;
+
+    return (
+      username &&
+      walletAddress &&
+      (`@${username}`.toLowerCase().includes(filterValue.toLowerCase()) ||
+        walletAddress.toLowerCase().includes(filterValue.toLowerCase()) ||
+        user.id.toLowerCase().includes(filterValue.toLowerCase()))
```

Comment:
I'm going to nitpick some more :P

Why call `filterValue.toLowerCase()` three times ? It won't change, so why not assign it to a variable, and use that for checks ?

```suggestion
      (`@${username}`.toLowerCase().includes(valueToLoweCase) ||
        walletAddress.toLowerCase().includes(valueToLoweCase) ||
        user.id.toLowerCase().includes(valueToLoweCase))
```